aviatrix/data_source_aviatrix_network_domains.go

```
@@ -0,0 +1,120 @@
+						},
+						"tgw_name": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "AWS TGW name.",
+						},
+						"route_table_id": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Route table id.",
+						},
+						"account": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Access Account name.",
+						},
+						"cloud_type": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Type of cloud service provider.",
+						},
+						"region": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Region of cloud provider.",
+						},
+						"intra_domain_inspection": {
+							Type:        schema.TypeBool,
+							Computed:    true,
+							Description: "Firewall inspection for traffic within one Security Domain.",
+						},
+						"egress_inspection": {
+							Type:        schema.TypeBool,
+							Computed:    true,
+							Description: "Egress inspection is enable or not.",
+						},
+						"inspection_policy": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Inspection policy name.",
+						},
+						"intra_domain_inspection_name": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Intra domain inspection name.",
+						},
+						"egress_inspection_name": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Egress inspection name.",
+						},
+						"type": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Type of Network Domain.",
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceAviatrixNetworkDomainsRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*goaviatrix.Client)
+
+	domainList, err := client.GetAllSecurityDomains()
+	if err != nil {
+		return err
```

We don't want to just return the error at this level. You need to add context to the error. Maybe something like `could not get Aviatrix Network Domains: err...`

---

have fixed